### DISCUSSION TITLE
Is commission calculated when entering and exiting a futures trade

### DISCUSSION THREAD

**Discussion by Chris2L** [kernc/backtesting.py#888] [Q&A]
I would just like to confirm if the commissions are taken into account when entering and exiting a futures trade.  According to Binance's FAQ, the commission (actually different for the market maker and market taker, and would be a nice feature to have) is taken when opening a contract and when closing a contract.  Is this the same for other exchanges?  For a worst-case analysis, should you just make the commission 2x of the market taker?

https://www.binance.com/en/support/faq/binance-futures-fee-structure-fee-calculations-360033544231

Thank you for the great library.  It's been a lot of help getting my feet wet in trading strategies.

**Comment by pazooki** [comment]
I have the same question. does the commission we set in Backtest() cause the fee to be applied once for entering and second time for exit? (since they are two different trades) or do you apply it only on exit/closing of position?

**Reply by pazooki** [reply]
Also having fees in `stat['_trades']` for each trade is very helpful

**Comment by copyright2010** [comment]
![image](https://github.com/kernc/backtesting.py/assets/24411085/b5c4e0a4-162d-4cf1-85ea-24324d035c6d)

**Comment by tsunamilx** [comment]
From what I observed, the commission is only applied for entries, and it is actually more like spread instead of commission, because it adjusts the entry price by the commission value you set (see `backtesting._Broker._adjusted_price`) and only for entry price.

I think this PR https://github.com/kernc/backtesting.py/pull/1119 makes more sense for the commission, @kernc can you take a look at it?
